code/API_definitions/geofencing.yaml

```diff
@@ -0,0 +1,785 @@
+      type: object
+      required:
+        - id
+        - source
+        - type
+        - specversion
+        - time
+      properties:
+        id:
+          description: Identifier of this event, that must be unique in the source context.
+          type: string
+        source:
+          $ref: '#/components/schemas/Source'
+        type:
+          $ref: '#/components/schemas/Type'
+        specversion:
+          description: Version of the specification to which this event conforms (must be 1.0 if it conforms to cloudevents 1.0.2 version)
+          type: string
+          enum:
+            - '1.0'
+        datacontenttype:
+          description: 'media-type that describes the event payload encoding, must be "application/json" for CAMARA APIs'
+          type: string
+          enum:
+            - application/json
+        time:
+          $ref: '#/components/schemas/EventTime'
+      discriminator:
+        propertyName: 'type'
+        mapping:
+          org.camaraproject.geofencing.v0.area-left: '#/components/schemas/EventAreaLeft'
+          org.camaraproject.geofencing.v0.area-entered: '#/components/schemas/EventAreaEntered'
+          org.camaraproject.geofencing.v0.subscription-ends: '#/components/schemas/EventSubscriptionEnds'
+
+    EventAreaLeft:
+      description: event structure for event when the device leaves the area
+      allOf:
+        - $ref: '#/components/schemas/CloudEvent'
+        - type: object
+          properties:
+            data:
+              $ref: '#/components/schemas/AreaLeft'
+
+    EventAreaEntered:
+      description: event structure for event when the device enters the area
+      allOf:
+        - $ref: '#/components/schemas/CloudEvent'
+        - type: object
+          properties:
+            data:
+              $ref: '#/components/schemas/AreaEntered'
+
+    EventSubscriptionEnds:
+      description: event structure for event subscription ends
+      allOf:
+        - $ref: '#/components/schemas/CloudEvent'
+        - type: object
+          properties:
+            data:
+              $ref: '#/components/schemas/SubscriptionEnds'
+
+    AreaLeft:
+      description: Event detail structure for area-left event
+      type: object
+      required:
+        - device
+        - area
+      properties:
+        device:
+          $ref: '#/components/schemas/Device'
+        area:
+          $ref: '#/components/schemas/Area'
+
+    AreaEntered:
+      description: Event detail structure for area-entered event
+      type: object
+      required:
+        - device
+        - area
+      properties:
+        device:
+          $ref: '#/components/schemas/Device'
+        area:
+          $ref: '#/components/schemas/Area'
+
+    SubscriptionEnds:
+      description: Event detail structure for SUBSCRIPTION_ENDS event
+      type: object
+      required:
+        - device
+        - terminationReason
+      properties:
+        device:
```

Comment:
Not sure if we need to add device to this one. And in that case, area would make sense as well, as we may have several subscriptions for same device but different areas. But in case of subscription termination it may be enough with subscriptionId and terminationReaso, wdyt @bigludo7 ?

Reply:
@jlurien For me it depends if GET /subscription is still possible for terminated one.

If yes we do no need device & area - if not then probably better to have all this information in the termination event. As we do not enforce this persistance probably good to provide this information.

Reply:
Okay then I will provide "area", too in the case persistence is not planned.